### PR TITLE
[SIDE-984] Add Initial C# LDK 

### PR DESCRIPTION
**Pull request by richseviora**
This PR adds the initial work on the C# LDK:

- Stands up the Loop Server (with GRPC Plugin and Stdio services).
- Receives Loop Start message.
- Implements the Clipboard Service.
- Implements the Whisper Markdown and Whisper Confirm messages.
- Implements the Logger.
- An example program that echoes the Clipboard content in a whisper.

This is not the complete LDK implementation, outstanding, and for implementation in future PRs are:

- All services excluding the Clipboard service.
- Implementing the Whisper service's WhisperForm message.
- Documentation.
- Other work necessary for publishing to NuGet.
- Unit Tests

### CODE REVIEW

**Review by BillyMurrayOlive — APPROVED** [review]
You got this working fast. Glad you know what you're doing.

File: ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Whispers/IWhisperService.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace OliveHelpsLDK.Whispers
+{
+    public interface IWhisperService
+    {
+        Task MarkdownAsync(Whispers.WhisperMarkdown message, CancellationToken cancellationToken = default);
+        Task<bool> ConfirmAsync(Whispers.WhisperConfirm message, CancellationToken cancellationToken = default);
+        // TODO: Implement form.
```
Comment (iangallOlive) [inline ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Whispers/IWhisperService.cs:10]: Do we have a card for this?

**Review by iangallOlive — APPROVED** [review]

File: ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Whispers/IWhisperService.cs (hunk 1 above)
Comment (richseviora) [reply ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Whispers/IWhisperService.cs:10]: We do now! SIDE-1007

**Comment by richseviora** [comment]
@dmorrison-olive I've merged this in but I'd still love to get your review.

**Comment by richseviora** [comment]
> You got this working fast. Glad you know what you're doing.

Hey I only _kinda_ know what I'm doing maybe #notAnCSharpExpert

**Comment by dmorrison-olive** [comment]
> @dmorrison-olive I've merged this in but I'd still love to get your review.

Sounds good! I have it on my list to take some time and review this. Should help me to better understand the LDK and overall Olive Helps architecture also. 👍

**Pull request merged**